### PR TITLE
Mailchimp decorator enables corresponding category wizard and legends

### LINKED ISSUE #3826: Mailchimp decorator doesn't apply the styles on the category and it should
Right now the wizard seems broken after connecting mailchimp data.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #3826.

@Kartones can you please take a look? I added some inline comments. Thanks!

### CODE REVIEW

File: app/models/layer.rb [hunk 1]
```diff
@@ -164,13 +164,23 @@ def get_presenter(options, configuration)
     CartoDB::Layer::Presenter.new(self, options, configuration)
   end
 
-  def set_style_options(cartocss_style)
+  def set_style_options(cartocss_style, opts = {})
```
Comment (alonsogarciapablo) [inline app/models/layer.rb]: These 3 methods are only being used by the `MailchimpDecorator`. I'd be happy to create a more generic method like `set_option(name, value)` and do something like `layer.set_option('tile_style', cartocss)` from the `MailchimpDecorator`. Thoughts?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4017/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4018/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4019/))

File: app/models/layer.rb (hunk 1 above)
Comment (Kartones) [inline app/models/layer.rb]: definetly yes!

I just added that as a first approach as current Layer model is quite barebones (as frontend JS is usually who controls layer data), so any addition is more than welcome!

**Comment by Kartones** [comment]
:+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4027/))

**Comment by alonsogarciapablo** [comment]
Thanks @Kartones. Please take a look at [this commit](https://github.com/CartoDB/cartodb/commit/65332b5457b5e54bac7caccc81ca251839c65165) that introduces `Layer#set_option` to have a more generic way of setting options. Thanks!

**Pull request merged**